glean-core/src/database/mod.rs

```diff
@@ -843,20 +880,49 @@ mod test {
                 _ => panic!("Unexpected data found"),
             }
         };
-
         db.iter_store_from(Lifetime::Ping, test_storage, None, &mut snapshotter);
         assert_eq!(1, found_metrics, "We only expect 1 Lifetime.Ping metric.");
 
-        // Make sure data was **not** persisted with rkv.
-        let store: SingleStore = unwrap_or!(
-            db.rkv
-                .open_single(Lifetime::Ping.as_str(), StoreOptions::create()),
-            panic!()
-        );
-        let reader = unwrap_or!(db.rkv.read(), panic!());
-        assert!(store
-            .get(&reader, &test_metric_id)
-            .unwrap_or(None)
-            .is_none());
+        {
+            // Make sure data **wasn't** persisted with rkv.
+            let store: SingleStore = db.rkv
+                .open_single(Lifetime::Ping.as_str(), StoreOptions::create())
+                .unwrap();
+            let reader = db.rkv.read().unwrap();
+            assert!(store
+                .get(&reader, format!("{}#{}", test_storage, test_metric_id))
+                .unwrap_or(None)
+                .is_none());
+        }
+
+        // Attempt to persist data.
+        db.persist_ping_lifetime_data().unwrap();
+
+        {
+            // Make sure that data **was** persisted with rkv.
+            let store: SingleStore = db.rkv
+                .open_single(Lifetime::Ping.as_str(), StoreOptions::create())
+                .unwrap();
+            let reader = db.rkv.read().unwrap();
+            assert!(store
+                .get(&reader, format!("{}#{}", test_storage, test_metric_id))
```

Comment:
This is essentially what `get_storage_key` does, but that is not a public method.